src/Runner.Listener/JobDispatcher.cs

```diff
@@ -507,7 +507,20 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, string orc
                                 {
                                     detailInfo = string.Join(Environment.NewLine, workerOutput);
                                     Trace.Info($"Return code {returnCode} indicate worker encounter an unhandled exception or app crash, attach worker stdout/stderr to JobRequest result.");
-                                    await LogWorkerProcessUnhandledException(message, detailInfo);
+
+                                    var jobServer = HostContext.GetService<IJobServer>();
```

Comment:
these are pulled out from the `LogWorkerProcessUnhandledException()` since I need to re-use `jobServer` later.